lib/telegraf.py

```diff
@@ -103,8 +104,13 @@ def _write_http_output_config(http_config):
     username = http_config.get('username')
     password = http_config.get('password')
     if username is not None:
```

Comment:
This wasn't changed here, but it's better to check for `if not username`.

If username happens to be the empty string, then the current statement (`is not None`) will evaluate to `True`, which is most likely not what you want.